pkg/distributor/ha_tracker.go

```diff
@@ -212,10 +212,47 @@ func newHATracker(cfg HATrackerConfig, limits haTrackerLimits, reg prometheus.Re
 		t.client = client
 	}
 
-	t.Service = services.NewBasicService(nil, t.loop, nil)
+	t.Service = services.NewBasicService(t.syncHATrackerStateOnStart, t.loop, nil)
 	return t, nil
 }
 
+func (h *haTracker) syncHATrackerStateOnStart(ctx context.Context) error {
+	if !h.cfg.EnableHATracker {
+		return nil
+	}
+	// haTracker holds HATrackerConfig with the prefix.
+	keys, err := h.client.List(ctx, "")
+	if err != nil {
+		return err
+	}
+
+	if len(keys) == 0 {
+		level.Warn(h.logger).Log("msg", "sync HA state on start: no keys for HA tracker prefix", "err", err)
+		return nil
+	}
+
+	for i := 0; i < len(keys); i++ {
+		if ctx.Err() != nil {
+			return fmt.Errorf("syncing HA tracker state on startup: %w", context.Cause(ctx))
+		}
+
+		val, err := h.client.Get(ctx, keys[i])
+		if err != nil {
+			level.Warn(h.logger).Log("msg", "sync HA state on start: failed to get replica value", "key", keys[i], "err", err)
```

Comment:
do we want to fail startup here or just continue? The `WatchKey` would ignore an unparsable value. Swallowing the error would also make us resilient to random values in the KV store (maybe it's being shared between multiple applications and someone forgot to configure a prefix for Mimir)

Reply:
My thought here was that if we have received the key, we should have a respective value, otherwise we should fail 🤔 

However I think it makes sense to continue and not fail here for resilient purposes as you mentioned. I will updated it